hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetInstanceImpl.java

```diff
@@ -51,6 +51,7 @@ public JetInstanceImpl(HazelcastInstanceImpl hazelcastInstance, JetConfig config
         this.nodeEngine = hazelcastInstance.node.getNodeEngine();
         this.config = config;
         this.shutdownHookThread = shutdownHookThread(nodeEngine);
+        Runtime.getRuntime().addShutdownHook(shutdownHookThread);
```

Comment:
the hook shouldn't be created here as JetInstanceImpl is just a wrapper and can be created several times. It should be created in JetService instead.